local/include/core/verification.h

```diff
@@ -359,15 +436,25 @@ class RuleCheck
    *
    * @return The name of the attribute operated on
    */
-  swoc::TextView
-  target_type() const
-  {
-    if (_is_field) {
-      return "Field Name";
-    } else {
-      return "URI Part";
-    }
-  }
+  swoc::TextView target_type() const;
+
+  /** Returns the not/nocase attributes the test operates on as a string
+   *
+   * @return The string holding the not/nocase attributes, if any
+   */
+  swoc::TextView get_subtype() const;
+
+  /** Returns a Success or Violation string depending on the test attributes
+   *
+   * @return The string Success or Violation
+   */
+  swoc::TextView invert_result(bool success) const;
+
+  /** Returns !input if is_inverted, otherwise input
+   *
+   * @return The resulting bool from the above operation
+   */
+  bool do_invert(bool input) const;
```

Comment:
I suggest something like `invert_if_applicable` or something like that. `do_invert` seems to imply it will simply invert the parameter.